warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IngestTypePruningVisitor.java

```diff
@@ -240,7 +240,7 @@ private Set<String> visitMarker(QueryPropertyMarker.Instance instance, JexlNode
         JexlNode source = node.jjtGetChild(1);
         Set<String> dts = (Set<String>) visit(source, data);
 
-        if (source.jjtGetNumChildren() == 0) {
+        if (source.jjtGetParent() == null) {
             pruneNodeFromParent(source);
```

Comment:
Do we need this and would if fail?

Reply:
We do not need this extra method call. It won't fail, there's a safeguard that does an early null check.